src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -37,6 +37,9 @@
  */
 abstract class AbstractObjectNormalizer extends AbstractNormalizer
 {
+    /** @var array<string, string|null> */
+    private array $discriminatorCache = [];
```

Comment:
there's already a cache in ClassDiscriminatorFromClassMetadata to maybe we don't need this one?
maybe this could mean we can remove the added protected method and inline the logic instead?

Reply:
Indeed, the less protected method we have, the better. It kind of duplicate the code a bit but IMHO it acceptable.